File: 練習/3/Cuboid.cpp
#include "Cuboid.h"

Cuboid::Cuboid() {

}
Cuboid::Cuboid(double, double, double) {

}
Cuboid::Cuboid(double l, double w, double h, string c)
{
	setLength(l);
	setWidth(w);
	setHeight(h);
	setColor(c);
}

void Cuboid::setLength(double l)
{
	this->length = l;
}

void Cuboid::setWidth(double w)
{
	this->width = w;
}

void Cuboid::setHeight(double h)
{
	this->height = h;
}

double Cuboid::getLength()
{
	return length;
}

double Cuboid::getWidth()
{
	return width;
}

double Cuboid::getHeight()
{
	return height;
}

double Cuboid::getVolume()
{
	return (length * width * height);
}

string Cuboid::toString()
{
	return "Cuboid:";
}

File: 練習/6/Cuboid.py
from Shape import Shape

class Cuboid(Shape):
    def __init__(self, length, width, height, color):
        self.__length = length
        self.__width = width
        self.__height = height
        super().setColor(color)
        
    def setLength(self, length):
        self.__length = length
        
    def setWidth(self, width):
        self.__width = width
        
    def setHeight(self, height):
        self.__height = height
        
    def getLength(self):
        return self.__length
    
    def getWidth(self):
        return self.__width
    
    def getHeight(self):
        return self.__height
        
    def getVolume(self):
        volume = self.getLength() * self.getWidth() * self.getHeight()
        return round(volume, 1)
    
    def __str__(self):
        return "Cuboid:"
        